blocks/core/ff_module/ff_module-task-event/_src/MarkAndGradeTaskEvent.js

```diff
@@ -0,0 +1,64 @@
+'use strict';
+
+var React = require('react');
+var TaskEventBase = require('./TaskEventBase'),
+    taskEventWithOptionalMessage = require('./taskEventWithOptionalMessage');
+
+module.exports.defaultState = React.createClass({
+    displayName: 'MarkAndGradeTaskEventDefault',
+    render: defaultState
+});
+
+module.exports.deletedState = React.createClass({
+    displayName: 'MarkAndGradeTaskEventDeleted',
+    render: taskEventWithOptionalMessage(function(props){
+        return getStatusText('deleted a ', props.event)+ '.';
+    })
+});
+
+function defaultState(){
+    var event = this.props.event,
+        messageText = event.message,
+        markText = getMarkText(event),
+        gradeText = event.grade;
+
+    var statusText = getStatusText(' added a ', event);
+
+    var mark = markText ? <span className="ff_module-task-event__mark">{markText}</span> : null,
+        grade = gradeText ? <span className="ff_module-task-event__grade">{gradeText}</span> : null,
+        sep = (markText && gradeText) ? ', ' : '';
+
+    var markAndGrade = (mark || grade) ? <p className="ff_module-task-event__mark-and-grade">{mark}{sep}{grade}</p> : null;
+
+    var message = messageText ? <p className="ff_module-task-event__message">{messageText}</p> : null;
+
+    return  <TaskEventBase event={event}>
+                <p className="ff_module-task-event__author-action">{event.author.name}{statusText}:</p>
+                {markAndGrade}
+                {message}
+            </TaskEventBase>
+}
+
+function getStatusText(base, event) {
+    switch (true) {
+        case !!(event.mark && !event.grade):
+            return base + 'mark';
+            break;
+        case !!(!event.mark && event.grade):
+            return base + 'grade';
+            break;
+        case !!(event.mark && event.grade):
+            return base + 'mark and grade';
+            break;
```

Comment:
FWIW, maybe nesting the `if`s would help?
>
>
```

if (event.mark) {

  if (event.grade) {

    return base + 'mark and grade';

  } else {

    return base + 'mark';

  }

}  else if (event.grade) {
  return base + 'grade';
}

```



-- 


[image: Firefly] <http://fireflysolutions.co.uk/>
Robin Message - Head of Technology
Firefly Learning LLP, 1 Lyric Square, London W6 0NB, UK

Support: +44 (0)20 7112 9362 <+44+(0)20+7112+9362> | Sales: +44 (0)20 8133
4415 <+44+(0)20+8133+4415> | fireflylearning.com | @fireflyteam
<http://twitter.com/fireflyteam>
Registered in England and Wales with partnership number OC336950.
VAT number: 981 8080 93.